From bde689f6981b4847a876648062be7c95c18c5e5a Mon Sep 17 00:00:00 2001
From: Jim Meyering <meyering@redhat.com>
Date: Thu, 11 Mar 2010 19:01:41 +0100
Subject: [PATCH] libparted: remove "HIGHLY EXPERIMENTAL" warning for
 >512B-sector devices

* libparted/arch/linux.c (_device_set_sector_size): Remove the
"HIGHLY EXPERIMENTAL" warning, now that support for >512-byte sectors
seems to be usable.  Suggested by Colin Watson.
* NEWS (Changes in behavior): Note this
---
 NEWS                   |  6 ++++++
 libparted/arch/linux.c | 10 ----------
 2 files changed, 6 insertions(+), 10 deletions(-)

diff --git a/NEWS b/NEWS
index 70f8d2e..d8ced7d 100644
--- a/NEWS
+++ b/NEWS
@@ -57,6 +57,12 @@ GNU parted NEWS                                    -*- outline -*-
   Add support for prep flag to GPT to select PowerPC Reference Platform
   boot partition type.
 
+** Changes in behavior
+
+  libparted no longer issues an exception/warning about >512-byte
+  sector support being "HIGHLY EXPERIMENTAL".  It seems to have matured
+  enough to be usable at least with GPT and MSDOS partition tables.
+
 ** Bug fixes
 
   libparted: raise the limit on the maximum start sector and the
diff --git a/libparted/arch/linux.c b/libparted/arch/linux.c
index a2793e1..c370bb6 100644
--- a/libparted/arch/linux.c
+++ b/libparted/arch/linux.c
@@ -695,16 +695,6 @@ _device_set_sector_size (PedDevice* dev)
                 dev->sector_size = PED_SECTOR_SIZE_DEFAULT;
         }
 #endif
-
-        if (dev->sector_size != PED_SECTOR_SIZE_DEFAULT) {
-                ped_exception_throw (
-                        PED_EXCEPTION_WARNING,
-                        PED_EXCEPTION_OK,
-                        _("Device %s has a logical sector size of %lld.  Not "
-                          "all parts of GNU Parted support this at the moment, "
-                          "and the working code is HIGHLY EXPERIMENTAL.\n"),
-                        dev->path, dev->sector_size);
-        }
 }
 
 static int
-- 
1.9.0

